llvm/lib/Target/AMDGPU/SIInstructions.td

```diff
@@ -1934,6 +1934,26 @@ def : GCNPat <
   (V_MOV_B32_e32 (f16 (bitcast_fpimm_to_i32 $imm)))
 >;
 
+def : GCNPat <
+  (VGPRImm<(i64 imm)>:$imm),
+  (V_MOV_B64_PSEUDO imm:$imm)
+>;
+
+def : GCNPat <
+  (VGPRImm<(f64 fpimm)>:$imm),
+  (V_MOV_B64_PSEUDO (f64 (bitcast_fpimm_to_i64 $imm)))
+>;
+
+def : GCNPat <
+  (i64 imm:$imm),
+  (S_MOV_B64_IMM_PSEUDO imm:$imm)
```

Comment:
AIUI the whole point of S_MOV_B64_IMM_PSEUDO is that unlike S_MOV_B64 it can be used with any 64-bit imm, not just encodable ones. But here we're only going to select it for encodable constants after all. It would be nice to have a comment explaining all that. I guess the same applies to V_MOV_B64_PSEUDO.

Reply:
Added comment.